src/libsystemd/sd-journal/sd-journal.c

```diff
@@ -2805,19 +2810,20 @@ _public_ int sd_journal_get_cutoff_monotonic_usec(sd_journal *j, sd_id128_t boot
                         continue;
 
                 if (found) {
-                        if (from)
-                                *from = MIN(fr, *from);
-                        if (to)
-                                *to = MAX(t, *to);
+                        from = MIN(ff, from);
+                        to = MAX(tt, to);
```

Comment:
The logic is slightly different here now right? Before, if the return values `*from`/`*to` were set, they could potentially have overshadowed any `fr`/`t` found while iterating on the hashmap (ie: if `*from` was smaller than all the values found in the iteration, it would have been returned unchanged). Now they will be ignored. Was the previous behavior intentional, and is the change intentional too?

Reply:
Actually in the old code we'd override what's in the return parameters if at least one file was found (that's because `found` is initially false, in which case we'll override whatever is written into these values). If no file was found we'd leave the values untouched, but still succeed. And that is really a bug: we should initialized all return params on sucess and leave them untouched on failure, as per coding style. Thus, with the new code we follow the coding style here, as when we don#t have any file open we'll now reset the return parameters to UINT64_MAX.

Reply:
Ah yes of course, missed that there's the first iteration special casing, thanks